pipeline/transformer/transformer_mets/src/main/scala/uk/ac/wellcome/platform/transformer/mets/service/MetsTransformerWorkerService.scala

```diff
@@ -1,33 +1,34 @@
 package uk.ac.wellcome.platform.transformer.mets.service
 
+import scala.concurrent.Future
 import akka.Done
 import grizzled.slf4j.Logging
+
 import uk.ac.wellcome.bigmessaging.BigMessageSender
 import uk.ac.wellcome.json.JsonUtil._
 import uk.ac.wellcome.messaging.sns.{NotificationMessage, SNSConfig}
 import uk.ac.wellcome.messaging.sqs.SQSStream
 import uk.ac.wellcome.mets_adapter.models.MetsLocation
 import uk.ac.wellcome.models.work.internal.TransformedBaseWork
-import uk.ac.wellcome.platform.transformer.mets.store.TemporaryCredentialsStore
-import uk.ac.wellcome.platform.transformer.mets.transformer.MetsXml
-import uk.ac.wellcome.storage.store.VersionedStore
-import uk.ac.wellcome.storage.{Identified, Version}
+import uk.ac.wellcome.platform.transformer.mets.transformer.MetsXmlTransformer
+import uk.ac.wellcome.storage.store.{Readable, VersionedStore}
+import uk.ac.wellcome.storage.{Identified, ObjectLocation, Version}
 import uk.ac.wellcome.typesafe.Runnable
 
-import scala.concurrent.Future
-
 class MetsTransformerWorkerService(
   msgStream: SQSStream[NotificationMessage],
   messageSender: BigMessageSender[SNSConfig, TransformedBaseWork],
   adapterStore: VersionedStore[String, Int, MetsLocation],
-  metsXmlStore: TemporaryCredentialsStore[String]
+  metsXmlStore: Readable[ObjectLocation, String]
 ) extends Runnable
     with Logging {
 
   type Result[T] = Either[Throwable, T]
 
   val className = this.getClass.getSimpleName
 
+  val xmlTransformer = new MetsXmlTransformer(metsXmlStore)
```

Comment:
Nitpicking here but why don't we inject the `xmlTransformer ` instead of `metsXmlStore`?

Reply:
I thought about this, and chose the the former as it means the creator `MetsTransformerWorkerService` only needs to know about the store (as opposed to both the store and the transformer). It's not a major issue for me either, but think it is slightly cleaner personally keeping the transformer as something internal to this class.